CMakeLists.txt

```diff
@@ -604,6 +611,7 @@ if (MSVC)
   endif()
 endif ()
 
+# FIXME: refactor/cleanup check below
```

Comment:
Can we not do this now?

Reply:
Fore sure, I won't do this now.

Reply:
I think we can remove the previously existing compiler checks safely within this PR, at least the checks for specific compiler versions.
I am also happy if we do it seperately.